src/java.base/share/classes/java/util/HexDigits.java

```diff
@@ -85,10 +47,10 @@ final class HexDigits implements Digits {
         short[] digits = new short[16 * 16];
 
         for (int i = 0; i < 16; i++) {
-            short hi = (short) ((i < 10 ? i + '0' : i - 10 + 'a') << 8);
+            short hi = (short) (i < 10 ? i + '0' : i - 10 + 'a');
 
             for (int j = 0; j < 16; j++) {
-                short lo = (short) (j < 10 ? j + '0' : j - 10 + 'a');
+                short lo = (short) ((j < 10 ? j + '0' : j - 10 + 'a') << 8);
                 digits[(i << 4) + j] = (short) (hi | lo);
```

Comment:
> A description of the DIGITS array layout and indexing is needed; it will help with the comprehension of the masking and shifts in the packDigits methods.

The DIGITS table is missing a description of its indexing.